api-report/tree2.api.md

```diff
@@ -1303,7 +1303,11 @@ const MarkType: {
 export type MatchPolicy = "subtree" | "path";
 
 // @alpha
-export type MemoizedIdRangeAllocator = (revision: RevisionTag | undefined, startId: ChangesetLocalId, count?: number) => IdRange[];
+export interface MemoizedIdRangeAllocator {
+    allocate(revision: RevisionTag | undefined, startId: ChangesetLocalId, count?: number): IdRange[];
+    mint(count?: number): IdRange[];
+    nextId: number;
```

Comment:
While the implementation will need to contain the nextId like this, I don't think there is any incentive to make it public in the API (and certainly not mutable, since someone overwriting this could make the allocator fail to uphold its guarantees).